Refactor: Xor8Builder::populate(`IntoIterator`)

drmingdrmer opened the pull request:
## Changelog

##### Refactor: Xor8Builder::populate(`IntoIterator`)


##### M  src/bin/perf.rs




- Improvement

drmingdrmer committed:
Refactor: Xor8Builder::populate(`IntoIterator`)

prataprc reviewed (changes requested):
(a) There are come compilation warnings.
(b) Also `cargo doc` seem to have some dangling references to symbol-names that are now modified.
(c) Let us do typedef for BuildHasherDefault and mark it as deprecated.

```rust
#[deprecated(since = "0.5.2", note = "Renamed to DefaultHasherBuilder")]
#[allow(dead_code)]
pub type BuildHasherDefault = DefaultHasherBuilder;
```

drmingdrmer committed:
Refactor: remove unused import

drmingdrmer committed:
CI: add lint command to Makefile and github-action

drmingdrmer committed:
Refactor: fix lint

drmingdrmer committed:
Refactor: fix broken link in doc

drmingdrmer commented:
> (a) There are come compilation warnings. (b) Also `cargo doc` seem to have some dangling references to symbol-names that are now modified. (c) Let us do typedef for BuildHasherDefault and mark it as deprecated.
> 
> ```rust
> #[deprecated(since = "0.5.2", note = "Renamed to DefaultHasherBuilder")]
> #[allow(dead_code)]
> pub type BuildHasherDefault = DefaultHasherBuilder;
> ```

Fixed. Have a check again, please.
